### PR TITLE
Add custom builders support

### PR DESCRIPTION
**Pull request by vespian**
This is very early WIP, just for sake of disusing stuff.

### CODE REVIEW

File: release/__init__.py [hunk 1]
```diff
@@ -454,24 +453,51 @@ def do_build_docker(name):
...
+
+    overlap = set(global_builders) & set(pkg_builders)
+    if overlap:
+        msg_fmt = "Pkg-defined builders overlap with global: `{}`"
+        raise BuildError(msg_fmt.format(overlap))
+
+    # FIXME: with python3.5 it is going to be: union_d12 = {**d1, **d2}
+    all_builders = global_builders.copy()
+    all_builders.update(pkg_builders)
+    for k in all_builders:
```
Comment (cmaloney) [inline release/__init__.py]:
```
for name, path in all_builders.items():
    do_build_docker(name, path)
```

File: release/__init__.py [hunk 2]
```diff
@@ -454,24 +453,51 @@ def do_build_docker(name):
...
+        do_build_docker(k, all_builders[k])
+
+
+def do_build_packages(cache_repository_url):
+    # TODO(cmaloney): Stop shelling out
```
Comment (cmaloney) [inline release/__init__.py]: nit: this old comment of mine can go away, we don't shell out anymore.

**Comment by cmaloney** [comment]
Seems reasonable. Definitely need a working downstream test before we ship it though.

File: release/__init__.py [hunk 3]
```diff
@@ -454,24 +453,51 @@ def do_build_docker(name):
...
+
+def _build_builders(package_store):
+    """Build all builder containers required to build packages
+    """
+    global_builders = _get_global_builders()
+    pkg_builders = package_store.builders
+
+    overlap = set(global_builders) & set(pkg_builders)
+    if overlap:
+        msg_fmt = "Pkg-defined builders overlap with global: `{}`"
```
Comment (vespian) [inline release/__init__.py]: :+1:

File: release/__init__.py (hunk 1 above)
Comment (vespian) [inline release/__init__.py]: True.

File: release/__init__.py (hunk 2 above)
Comment (vespian) [inline release/__init__.py]: done

**Comment by cmaloney** [comment]
🚢

**Pull request merged**